proto/substrait/algebra.proto

```diff
@@ -173,6 +173,8 @@ message JoinRel {
     // This join is useful for nested sub-queries where we need exactly one record in output (or throw exception)
     // See Section 3.2 of https://15721.courses.cs.cmu.edu/spring2018/papers/16-optimizer2/hyperjoins-btw2017.pdf
     JOIN_TYPE_SINGLE = 7;
+    JOIN_TYPE_MARK = 8;
+    JOIN_TYPE_RIGHT_SEMI = 9;
```

Comment:
The latest release now has the join type enums a bit normalized so you'll find RIGHT_SEMI here now.  MARK has not yet been added.

https://github.com/substrait-io/substrait/commit/bed84ecb6193c22bf2ff83dc3a391ec5a9a3aa68